cms/workflows/locks.py

```diff
@@ -0,0 +1,55 @@
+from typing import TYPE_CHECKING, Union
+
+from django.urls import reverse
+from django.utils.html import format_html
+from wagtail.locks import BaseLock
+
+from cms.bundles.permissions import user_can_manage_bundles
+
+if TYPE_CHECKING:
+    from django.utils.safestring import SafeString
+
+    from cms.users.models import User
+
+
+class PageInBundleReadyToBePublishedLock(BaseLock):
+    """A lock that is enabled when the page is in a bundle that is ready to be published."""
+
+    def for_user(self, user: "User") -> bool:
+        return (
+            getattr(self.object, "active_bundle", None) is not None
+            and self.object.active_bundle.is_ready_to_be_published
+        )
+
+    def get_message(self, user: "User") -> Union[str, "SafeString"]:
+        lock_message = ""
+        if user_can_manage_bundles(user):
+            return format_html(
+                "This page is included in a bundle that is ready to be published. You must revert the bundle "
+                "to <strong>Draft</strong> or <strong>In preview</strong> in order to make further changes. "
+                '<span class="buttons">'
+                '<a type="button" class="button button-small button-secondary" href="{url}">Manage bundle</a></span>',
+                base_message=lock_message,
+                url=reverse("bundle:edit", args=(self.object.active_bundle.pk,)),
+            )
```

Comment:
`lock_message` seems redundant here as it's an empty string and doesn't change

Reply:
ah, remnant from a different version. removed 49b3cfee